src/vs/workbench/common/viewlet.ts

```diff
@@ -5,4 +5,9 @@
 
 import {IComposite} from 'vs/workbench/common/composite';
 
-export interface IViewlet extends IComposite { }
+export interface IViewlet extends IComposite {
+	/**
+	 * Returns the minimal width needed to avoid any content horizontal truncation
+	 */
+	getOptimalWidth?(): number;
```

Comment:
Suggest to make this a non optional method and implement it by default in the base viewlet class.